DashDocs/src/utils/parse_example.jl

```diff
@@ -0,0 +1,131 @@
+
+const top_level_exprs = Set([:using, :import, :const])
+# pull up and return top level calls from Expr
+function top_level_popout!(e::Expr)
+    top_inds = findall(e.args) do arg
+        return isa(arg, Expr) && arg.head in top_level_exprs
+    end
+    result = e.args[top_inds]
+    deleteat!(e.args, top_inds)
+    return result
+end
+
+function check_is_download(right::Expr)
+    (right.head != :call || right.args[1] != :DataFrame || !isa(right.args[2], Expr)) && return nothing
+    (right.args[2].head != :call || right.args[2].args[1] != :urldownload) && return nothing
+    return string(right.args[2].args[2])
+end
+function check_is_rdataset(right::Expr)
+    (right.head != :call || right.args[1] != :dataset) && return nothing
+    return join(string.(right.args[2:end]), "|")
+end
+function check_dataset(right::Expr)
+    downlad_name = check_is_download(right)
+    return isnothing(downlad_name) ? check_is_rdataset(right) : downlad_name
+end
+function datasets_popout!(e::Expr)
+    for arg in e.args
+        !isa(arg, Expr) && continue
+        arg.head != :(=) && continue
+        (!isa(arg.args[1], Symbol)  || !isa(arg.args[2], Expr)) && continue
+        var = arg.args[1]
+        right = arg.args[2]
+        url = check_dataset(right)
+        isnothing(url) && continue
+        if !haskey(DATASETS, url)
+            DATASETS[url] = eval(right)
+        end
+        arg.args[2] = :(copy(DATASETS[$url]))
+    end
+end
+
+is_callbackcall(expr) = expr.head == :call && expr.args[1] == :callback!
+
+function is_callback(expr)
+  is_callbackcall(expr) && return true #callback!((x)->x, ....)
+  (expr.head == :do && is_callbackcall(expr.args[1])) && return true #do syntax
```

Comment:
this should be recursive?

Reply:
Yep, a reasonable remark. I added recursive `remove_callbacks!` at 52fa88a